website_compassion/controllers/my_account.py

```diff
@@ -296,23 +317,25 @@ def my_donations(self, form_id=None, **kw):
             ("date_invoice", "<", fields.Date.to_string(in_one_month))
         ])
 
+        last_completed_tax_receipt = partner.last_completed_tax_receipt
```

Comment:
Maybe this init is not really needed as we only use it on time (line 406)

Reply:
Yes the call is actually only needed in the creation of the dictionary used in the return statement. But when called directly in the dictionary it led to a strange behavior where last_completed_tax year was computed not only for the current user but also for other people. Maybe it has something to do with the state of self changing during the dictionary preparation. either way calling it before, as it is done here, solved the issue